File: Classes/BackgroundLayer.cpp


#include "BackgroundLayer.h"
#include "AtlasLoader.h"

BackgroundLayer::BackgroundLayer(){}
BackgroundLayer::~BackgroundLayer(){}

bool BackgroundLayer::init()
{
	if (!Layer::init())
	{
		return false;
	}

	time_t t = time(NULL);
	tm *lt = localtime(&t);
	int hour = lt->tm_hour;
	Sprite *background;
	if (hour >= 6 && hour <= 17)
	{
		background = Sprite::createWithSpriteFrame(AtlasLoader::getInstance()->getSpriteFrameByName("bg_day"));
	} else {
		background = Sprite::createWithSpriteFrame(AtlasLoader::getInstance()->getSpriteFrameByName("bg_night"));
	}
	background->setAnchorPoint(Point::ZERO);
	background->setPosition(Point::ZERO);
	this->addChild(background);

	return true;
}

float BackgroundLayer::getLandHeight()
{
	return Sprite::createWithSpriteFrame(AtlasLoader::getInstance()->getSpriteFrameByName("land"))->getContentSize().height;
}

File: src/app/views/BackgroundLayer.lua

local backgroundDayName = "bg_day"
local backgroundNightName = "bg_night"

local BackgroundLayer = class("BackgroundLayer", cc.Layer)

function BackgroundLayer:ctor()
	local atlasLoader = require("app.tools.AtlasLoader"):getInstance()

	local backgroundSprite = nil
	local timeFormat = os.date("*t")
	if(timeFormat.hour >= 6 and timeFormat.hour <= 17) then
		backgroundSprite = cc.Sprite:createWithSpriteFrame(atlasLoader:getSpriteFrameByName(backgroundDayName))
	else 
		backgroundSprite = cc.Sprite:createWithSpriteFrame(atlasLoader:getSpriteFrameByName(backgroundNightName))
	end
	backgroundSprite:setAnchorPoint({ x = 0, y = 0 })
	backgroundSprite:setPosition({ x = 0, y = 0 })
	self:addChild(backgroundSprite)
end

return BackgroundLayer
